src/WebJobs.Script/Eventing/ScriptEventManager.cs

```diff
@@ -11,7 +11,13 @@ public sealed class ScriptEventManager : IScriptEventManager, IDisposable
         private readonly Subject<ScriptEvent> _subject = new Subject<ScriptEvent>();
         private bool _disposed = false;
 
-        public void Publish(ScriptEvent scriptEvent) => _subject.OnNext(scriptEvent);
+        public void Publish(ScriptEvent scriptEvent)
+        {
+            if (!_disposed)
```

Comment:
_disposed gets set after _subject is disposed, meaning we can still get race conditions it seems

Reply:
True. I can set _disposed first to make sure we stop publishing once we've entered the Dispose() method. Alternatively I could lock around both of those code paths, but that doesn't seem necessary.

Reply:
I'm ok with making this a noop if we've been disposed if there is no better way, however it does seem that the true problem exists upstream. Someone is calling methods on a disposed object. Generally it's expected that calling methods on disposed objects will fail. This isn't a pattern we use in any other objects that I'm aware of.

Reply:
I don't think this is an issue in production (just tests) so I'll just back it out. The FileSystemWatcher scenario is more important as it affects Prod and is changing behavior.